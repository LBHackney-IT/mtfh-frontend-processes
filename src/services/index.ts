export { default as locale } from "./locale";
export { default as menu } from "./menu";
export * as processes from "./processes";
export * from "./config";
