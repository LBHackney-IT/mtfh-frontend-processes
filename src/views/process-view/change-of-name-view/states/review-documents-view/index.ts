export * from "./review-documents-view";
