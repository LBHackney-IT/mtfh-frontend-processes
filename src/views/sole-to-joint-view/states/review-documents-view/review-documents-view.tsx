import React, { useState } from "react";

import { Form, Formik } from "formik";

import { SoleToJointHeader } from "../../../../components";
import { AppointmentDetails } from "../../../../components/appointment-details/appointment-details";
import { AppointmentForm } from "../../../../components/appointment-form/appointment-form";
import { locale } from "../../../../services";
import { Trigger } from "../../../../services/processes/types";
import { IProcess } from "../../../../types";
import { DesBox, EligibilityChecksPassedBox } from "../shared";

import { Process, editProcess } from "@mtfh/common/lib/api/process/v1";
import {
  Button,
  Checkbox,
  Heading,
  StatusErrorSummary,
} from "@mtfh/common/lib/components";

const { views } = locale;
const { reviewDocuments } = views;
interface ReviewDocumentsViewProps {
  processConfig: IProcess;
  process: Process;
  mutate: () => void;
  optional?: any;
}

export const ReviewDocumentsView = ({
  processConfig,
  process,
  mutate,
  optional,
}: ReviewDocumentsViewProps) => {
  const [seenPhotographicId, setSeenPhotographicId] = useState<boolean>(false);
  const [seenSecondId, setSeenSecondId] = useState<boolean>(false);
  const [isNotInImmigrationControl, setIsNotInImmigrationControl] =
    useState<boolean>(false);
  const [seenProofOfRelationship, setSeenProofOfRelationship] = useState<boolean>(false);
  const [incomingTenantLivingInProperty, setIncomingTenantLivingInProperty] =
    useState<boolean>(false);

  const { states } = processConfig;
  const stateConfigs = {
    [states.documentsRequestedDes.state]: processConfig.states.documentsRequestedDes,
    [states.documentsRequestedAppointment.state]:
      processConfig.states.documentsRequestedAppointment,
    [states.documentsAppointmentRescheduled.state]:
      processConfig.states.documentsAppointmentRescheduled,
    [states.processClosed.state]: processConfig.states.processClosed,
  };
  const stateConfig = stateConfigs[process.currentState.state];
  const [globalError, setGlobalError] = useState<number>();

  return (
    <div data-testid="soletojoint-ReviewDocuments">
      <SoleToJointHeader processConfig={processConfig} process={process} />

      {globalError && (
        <StatusErrorSummary id="review-documents-global-error" code={globalError} />
      )}
      {!optional?.closeProcessReasonFinal && (
        <>
          <EligibilityChecksPassedBox />
          {(states.documentsRequestedDes.state === process.currentState.state ||
            process.previousStates.find(
              (previous) => previous.state === states.documentsRequestedDes.state,
            )) && <DesBox title={reviewDocuments.documentsRequested} />}

          <ReviewDocumentsAppointmentForm
            processConfig={processConfig}
            process={process}
            mutate={mutate}
            setGlobalError={setGlobalError}
          />

          <div style={{ paddingBottom: 35 }} />

          <Formik
            initialValues={{}}
            onSubmit={async () => {
              try {
                await editProcess({
                  id: process.id,
                  processTrigger: stateConfig.triggers.reviewDocuments,
                  processName: process?.processName,
                  etag: process.etag || "",
                  formData: {
                    seenPhotographicId,
                    seenSecondId,
                    isNotInImmigrationControl,
                    seenProofOfRelationship,
                    incomingTenantLivingInProperty,
                  },
                  documents: [],
                });
                mutate();
              } catch (e: any) {
                setGlobalError(e.response?.status || 500);
              }
            }}
          >
            {() => {
              return (
                <Form
                  noValidate
                  id="review-documents-form"
                  className="review-documents-form"
                >
                  <Heading variant="h4">{reviewDocuments.useFormBelow}</Heading>
                  <Checkbox
                    id="seen-photographic-id"
                    checked={seenPhotographicId}
                    onChange={() => setSeenPhotographicId(!seenPhotographicId)}
                    hint={reviewDocuments.seenPhotographicIdHint}
                  >
                    {reviewDocuments.seenPhotographicId}
                  </Checkbox>
                  <Checkbox
                    id="seen-second-id"
                    checked={seenSecondId}
                    onChange={() => setSeenSecondId(!seenSecondId)}
                    hint={reviewDocuments.seenSecondIdHint}
                  >
                    {reviewDocuments.seenSecondId}
                  </Checkbox>
                  <Checkbox
                    id="is-not-immigration-control"
                    checked={isNotInImmigrationControl}
                    onChange={() =>
                      setIsNotInImmigrationControl(!isNotInImmigrationControl)
                    }
                    hint={reviewDocuments.isNotInImmigrationControlHint}
                  >
                    {reviewDocuments.isNotInImmigrationControl}
                  </Checkbox>
                  <Checkbox
                    id="seen-proof-of-relationship"
                    checked={seenProofOfRelationship}
                    onChange={() => setSeenProofOfRelationship(!seenProofOfRelationship)}
                    hint={reviewDocuments.seenProofOfRelationshipHint}
                  >
                    {reviewDocuments.seenProofOfRelationship}
                  </Checkbox>
                  <Checkbox
                    id="incoming-tenant-living-in-property"
                    checked={incomingTenantLivingInProperty}
                    onChange={() =>
                      setIncomingTenantLivingInProperty(!incomingTenantLivingInProperty)
                    }
                    hint={reviewDocuments.incomingTenantLivingInPropertyHint}
                  >
                    {reviewDocuments.incomingTenantLivingInProperty}
                  </Checkbox>
                  <Button
                    type="submit"
                    disabled={
                      !seenPhotographicId ||
                      !seenSecondId ||
                      !isNotInImmigrationControl ||
                      !seenProofOfRelationship ||
                      !incomingTenantLivingInProperty
                    }
                    style={{ width: 222 }}
                  >
                    {locale.next}
                  </Button>
                </Form>
              );
            }}
          </Formik>
        </>
      )}
    </div>
  );
};

interface ReviewDocumentsAppointmentFormProps {
  processConfig: IProcess;
  process: Process;
  mutate: () => void;
  setGlobalError: any;
}

export const ReviewDocumentsAppointmentForm = ({
  processConfig,
  process,
  mutate,
  setGlobalError,
}: ReviewDocumentsAppointmentFormProps): JSX.Element => {
  const [needAppointment, setNeedAppointment] = useState<boolean>(false);

  const { states } = processConfig;
  return (
    <>
      <AppointmentDetails
        process={process}
        needAppointment={needAppointment}
        setNeedAppointment={setNeedAppointment}
        options={{
          requestAppointmentTrigger: Trigger.RequestDocumentsAppointment,
          rescheduleAppointmentTrigger: Trigger.RescheduleDocumentsAppointment,
          appointmentRequestedState: states.documentsRequestedAppointment.state,
          appointmentRescheduledState: states.documentsAppointmentRescheduled.state,
        }}
      />

      {(states.documentsRequestedDes.state === process.currentState.state ||
        needAppointment) && (
        <Checkbox
          id="condition"
          checked={needAppointment}
          onChange={() => setNeedAppointment(!needAppointment)}
        >
          {locale.views.reviewDocuments.checkSupportingDocumentsAppointment}
        </Checkbox>
      )}

      <AppointmentForm
        process={process}
        mutate={mutate}
        setGlobalError={setGlobalError}
        needAppointment={needAppointment}
        setNeedAppointment={setNeedAppointment}
        options={{
          buttonText: locale.bookAppointment,
          requestAppointmentTrigger: Trigger.RequestDocumentsAppointment,
          rescheduleAppointmentTrigger: Trigger.RescheduleDocumentsAppointment,
          appointmentRequestedState: states.documentsRequestedAppointment.state,
          appointmentRescheduledState: states.documentsAppointmentRescheduled.state,
        }}
      />
    </>
  );
};
