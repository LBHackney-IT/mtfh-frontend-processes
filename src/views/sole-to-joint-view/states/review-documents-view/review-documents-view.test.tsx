import React from "react";

import { patchProcessV1, render, server } from "@hackney/mtfh-test-utils";
import { fireEvent, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { locale, processes } from "../../../../services";
import {
  mockDocumentsRequestedAppointment,
  mockDocumentsRequestedDes,
  mockDocumentsRequestedDesAppointment,
} from "../../../../test-utils";
import { ReviewDocumentsView } from "./review-documents-view";

describe("review-documents-view", () => {
  beforeEach(() => {
    jest.resetModules();
  });

  test("it renders ReviewDocuments correctly on state DocumentsRequestedDes", async () => {
    const { container } = render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await expect(
      screen.findByText(locale.views.reviewDocuments.passedChecks),
    ).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(locale.views.reviewDocuments.viewInDes),
    ).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(locale.views.reviewDocuments.checkSupportingDocumentsAppointment),
    ).resolves.toBeInTheDocument();
    await expect(screen.findByText(locale.closeCase)).resolves.toBeInTheDocument();
    await expect(screen.getByText(locale.next)).toBeDisabled();
    expect(container).toMatchSnapshot();
  });

  test("it enables book appointment once checkbox is selected", async () => {
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
      />,
    );

    await expect(screen.queryByText(locale.bookAppointment)).toBeNull();
    const checkbox = screen.getByLabelText(
      locale.views.reviewDocuments.checkSupportingDocumentsAppointment,
    );
    fireEvent.click(checkbox);
    const bookAppointmentButton = await screen.findByText(locale.bookAppointment);
    expect(bookAppointmentButton).toBeDisabled();
  });

  test("it disables book appointment if date is not in future", async () => {
    const user = userEvent.setup();
    server.use(patchProcessV1("error", 500));
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
      />,
    );
    const checkbox = screen.getByLabelText(
      locale.views.reviewDocuments.checkSupportingDocumentsAppointment,
    );
    fireEvent.click(checkbox);
    await typeDateTime(user, "2000");
    const bookAppointmentButton = await screen.findByText(locale.bookAppointment);
    expect(bookAppointmentButton).toBeDisabled();
  });

  test("it displays an error if there's an issue with book appointment", async () => {
    const user = userEvent.setup();
    server.use(patchProcessV1("error", 500));
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
      />,
    );

    await userEvent.click(
      screen.getByLabelText(
        locale.views.reviewDocuments.checkSupportingDocumentsAppointment,
      ),
    );
    await typeDateTime(user, "2099");
    const bookAppointmentButton = await screen.findByText(locale.bookAppointment);
    expect(bookAppointmentButton).toBeEnabled();
    await user.click(bookAppointmentButton);
    await expect(
      screen.findByText("There was a problem with completing the action"),
    ).resolves.toBeInTheDocument();
  });

  test("it renders requested via des and book appointment", async () => {
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedDesAppointment}
        mutate={() => {}}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await expect(
      screen.findByText(locale.views.reviewDocuments.passedChecks),
    ).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(locale.views.reviewDocuments.viewInDes),
    ).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(locale.components.appointment.scheduled),
    ).resolves.toBeInTheDocument();
  });

  test("it displays change button if date is future", async () => {
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedAppointment({
          appointmentDateTime: "2099-10-12T08:59:00.000Z",
        })}
        mutate={() => {}}
      />,
    );

    await expect(
      screen.findByText(locale.components.appointment.scheduled),
    ).resolves.toBeInTheDocument();
    await expect(screen.findByText(locale.change)).resolves.toBeInTheDocument();
  });

  test("it displays reschedule button if date is past", async () => {
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedAppointment({
          appointmentDateTime: "2010-10-12T08:59:00.000Z",
        })}
        mutate={() => {}}
      />,
    );

    await expect(screen.queryByText(locale.bookAppointment)).toBeNull();
    const button = screen.getByText(locale.reschedule);
    fireEvent.click(button);
    const bookAppointmentButton = await screen.findByText(locale.bookAppointment);
    expect(bookAppointmentButton).toBeDisabled();
  });

  test("it enables next once all checkboxes are selected", async () => {
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
      />,
    );

    const nextButton = await screen.findByText(locale.next);
    await expect(nextButton).toBeDisabled();
    selectAllCheckBoxes();
    await expect(nextButton).toBeEnabled();
  });

  test("it displays an error if there's an issue with book appointment", async () => {
    server.use(patchProcessV1("error", 500));
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
      />,
    );

    const nextButton = await screen.findByText(locale.next);
    selectAllCheckBoxes();
    expect(nextButton).toBeEnabled();
    nextButton.click();
    await expect(
      screen.findByText("There was a problem with completing the action"),
    ).resolves.toBeInTheDocument();
  });

  test("it renders ReviewDocuments correctly on state DocumentsRequestedAppointment", async () => {
    const { container } = render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedAppointment({
          appointmentDateTime: "2099-10-12T08:59:00.000Z",
        })}
        mutate={() => {}}
      />,
      {
        url: "/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735",
        path: "/processes/soletojoint/:processId",
      },
    );
    await expect(
      screen.findByText(locale.views.reviewDocuments.passedChecks),
    ).resolves.toBeInTheDocument();
    await expect(screen.queryByText(locale.views.reviewDocuments.viewInDes)).toBeNull();
    await expect(
      screen.findByText(locale.components.appointment.scheduled),
    ).resolves.toBeInTheDocument();
    await expect(screen.findByText(/08:59 am/)).resolves.toBeInTheDocument();
    await expect(
      screen.findByText(/Monday 12th October 2099/),
    ).resolves.toBeInTheDocument();
    await expect(screen.findByText(locale.closeCase)).resolves.toBeInTheDocument();
    await expect(screen.getByText(locale.next)).toBeDisabled();
    expect(container).toMatchSnapshot();
  });

  test("it displays close case dialog on close case button click and closes on cancel", async () => {
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
      />,
    );

    await userEvent.click(screen.getByText(locale.closeCase));
    await expect(
      screen.findByText(locale.views.closeCase.reasonForRejection, { exact: false }),
    ).resolves.toBeInTheDocument();
    await userEvent.click(await screen.findByText(locale.cancel));
    await expect(
      screen.queryByText(locale.views.closeCase.reasonForRejection),
    ).toBeNull();
  });

  test("it displays close case view with reason and checkbox to confirm", async () => {
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
      />,
    );
    const reason = "Documents not provided";
    await userEvent.click(await screen.findByText(locale.closeCase));
    await expect(
      screen.findByText(locale.views.closeCase.reasonForRejection),
    ).resolves.toBeInTheDocument();
    await userEvent.type(
      screen.getByLabelText(`${locale.views.closeCase.reasonForRejection}*`),
      reason,
    );
    await userEvent.click(await screen.findByText(locale.confirm));
    await expect(
      screen.findByText(locale.views.reviewDocuments.soleToJointClosed),
    ).resolves.toBeInTheDocument();
    await expect(screen.findByText(reason)).resolves.toBeInTheDocument();
  });

  test("it closes case when checkbox is ticked and reason provided", async () => {
    server.use(patchProcessV1({}, 200));
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
      />,
    );

    const reason = "Documents not provided";
    await userEvent.click(await screen.findByText(locale.closeCase));
    await userEvent.type(
      screen.getByLabelText(`${locale.views.closeCase.reasonForRejection}*`),
      reason,
    );
    await userEvent.click(await screen.findByText(locale.confirm));
    await expect(screen.findByText(reason)).resolves.toBeInTheDocument();
    await expect(screen.findByText(locale.confirm)).resolves.toBeDisabled();
    await userEvent.click(
      screen.getByLabelText(locale.views.reviewDocuments.outcomeLetterSent),
    );
    await expect(screen.findByText(locale.confirm)).resolves.toBeEnabled();
    await userEvent.click(screen.getByText(locale.confirm));
    expect(
      screen.findByText(locale.views.reviewDocuments.confirmation),
    ).resolves.toBeInTheDocument();
  });

  test("it displays error if close case API call fails", async () => {
    server.use(patchProcessV1("", 500));
    render(
      <ReviewDocumentsView
        processConfig={processes.soletojoint}
        process={mockDocumentsRequestedDes}
        mutate={() => {}}
      />,
    );

    const reason = "Documents not provided";
    await userEvent.click(await screen.findByText(locale.closeCase));
    await userEvent.type(
      screen.getByLabelText(`${locale.views.closeCase.reasonForRejection}*`),
      reason,
    );
    await userEvent.click(await screen.findByText(locale.confirm));
    await userEvent.click(
      screen.getByLabelText(locale.views.reviewDocuments.outcomeLetterSent),
    );
    await userEvent.click(screen.getByText(locale.confirm));
    await expect(
      screen.findByText("There was a problem with completing the action"),
    ).resolves.toBeInTheDocument();
  });
});

function selectAllCheckBoxes() {
  fireEvent.click(screen.getByLabelText(locale.views.reviewDocuments.seenPhotographicId));
  fireEvent.click(screen.getByLabelText(locale.views.reviewDocuments.seenSecondId));
  fireEvent.click(
    screen.getByLabelText(locale.views.reviewDocuments.seenProofOfRelationship),
  );
  fireEvent.click(
    screen.getByLabelText(locale.views.reviewDocuments.isNotInImmigrationControl),
  );
  fireEvent.click(
    screen.getByLabelText(locale.views.reviewDocuments.incomingTenantLivingInProperty),
  );
}

async function typeDateTime(user, year) {
  await user.type(screen.getByPlaceholderText(/dd/i), "01");
  await user.type(screen.getByPlaceholderText(/mm/i), "01");
  await user.type(screen.getByPlaceholderText(/yy/i), year);
  await user.type(screen.getAllByPlaceholderText(/00/i)[0], "01");
  await user.type(screen.getAllByPlaceholderText(/00/i)[1], "01");
  await user.type(screen.getByPlaceholderText(/am/i), "am");
}
