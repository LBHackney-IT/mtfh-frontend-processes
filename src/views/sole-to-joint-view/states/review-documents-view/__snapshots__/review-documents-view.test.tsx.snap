// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`review-documents-view it renders ReviewDocuments correctly on state DocumentsRequestedAppointment 1`] = `
<div>
  <div
    data-testid="soletojoint-ReviewDocuments"
  >
    <h1
      class="lbh-heading-h1"
    >
      Sole tenant requests a joint tenure
    </h1>
    <h2
      class="lbh-heading-h2 entity-summary__tenure-heading"
    >
      <div>
        Tenure payment ref
         
        mt0xzvjp6g
      </div>
      <div>
        72251 Champlin Trafficway, 99647
      </div>
      <div>
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/person/a302fe28-e5c3-4c84-a3e6-cd575e65093f"
        >
          Global Web Consultant Alverta Denesik
        </a>
      </div>
    </h2>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Eligibility checks passed
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="padding-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735"
        >
          Show all eligibility checks
        </a>
      </div>
    </div>
    <div
      class="mtfh-box"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#0B0C0C"
            stroke-width="1.5"
          />
          <path
            d="M20.2251 12.7026C20.2251 11.1991 21.0627 10.4473 22.7378 10.4473C24.4129 10.4473 25.2505 11.1991 25.2505 12.7026C25.2505 13.4191 25.0386 13.9792 24.6147 14.3828C24.201 14.7764 23.5754 14.9731 22.7378 14.9731C21.0627 14.9731 20.2251 14.2163 20.2251 12.7026ZM25.0386 34H20.4219V17.0771H25.0386V34Z"
            fill="#0B0C0C"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Office appointment scheduled
        </h4>
      </div>
      <p
        class="lbh-body-m"
        style="margin-left: 60px;"
      >
        Date: 
        Monday 12th October 2099
        <br />
        Time: 
        08:59 am
      </p>
      <button
        class="govuk-link lbh-link"
        style="margin-left: 60px;"
        type="button"
      >
        Change
      </button>
    </div>
    <div
      style="padding-bottom: 35px;"
    />
    <form
      action="#"
      class="review-documents-form"
      id="review-documents-form"
      novalidate=""
    >
      <h4
        class="lbh-heading-h4"
      >
        Use the form below to record the documents you have checked:
      </h4>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          aria-describedby="seen-photographic-id-hint"
          class="govuk-checkboxes__input"
          id="seen-photographic-id"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="seen-photographic-id"
        >
          I confirm that an outcome letter has been sent to the resident
        </label>
        <span
          class="govuk-hint govuk-checkboxes__hint lbh-hint"
          id="seen-photographic-id-hint"
        >
          (for example: valid passport, driving licence)
        </span>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          aria-describedby="seen-second-id-hint"
          class="govuk-checkboxes__input"
          id="seen-second-id"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="seen-second-id"
        >
          I confirm I have seen a second form of ID (does not have to be photographic)
        </label>
        <span
          class="govuk-hint govuk-checkboxes__hint lbh-hint"
          id="seen-second-id-hint"
        >
          (for example: utility bill, bank statement, council letter)
        </span>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          aria-describedby="is-not-immigration-control-hint"
          class="govuk-checkboxes__input"
          id="is-not-immigration-control"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="is-not-immigration-control"
        >
          I confirm that the prospective tenant is not subject to immigration control under the Asylum And Immigration Act 1996
        </label>
        <span
          class="govuk-hint govuk-checkboxes__hint lbh-hint"
          id="is-not-immigration-control-hint"
        >
          (for example: passport, home office letter, embassy letter, immigration status document)
        </span>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          aria-describedby="seen-proof-of-relationship-hint"
          class="govuk-checkboxes__input"
          id="seen-proof-of-relationship"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="seen-proof-of-relationship"
        >
          I confirm I have seen proof of relationship to the existing tenant
        </label>
        <span
          class="govuk-hint govuk-checkboxes__hint lbh-hint"
          id="seen-proof-of-relationship-hint"
        >
          (for example: marriage or civil partner certificate)
        </span>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          aria-describedby="incoming-tenant-living-in-property-hint"
          class="govuk-checkboxes__input"
          id="incoming-tenant-living-in-property"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="incoming-tenant-living-in-property"
        >
          I confirm I have seen 3 separate documents proving the proposed tenant has been living at the property for a minimum of 12 months
        </label>
        <span
          class="govuk-hint govuk-checkboxes__hint lbh-hint"
          id="incoming-tenant-living-in-property-hint"
        >
          (for example: letter, utility bill, council tax bill)
        </span>
      </div>
      <button
        class="govuk-button lbh-button"
        disabled=""
        style="width: 222px;"
        type="submit"
      >
        Next
      </button>
    </form>
    <p
      class="lbh-body-m"
    >
      If the documents are not suitable and all avenues to obtain the right documents have been exhausted, then close the case.
    </p>
    <button
      class="govuk-button lbh-button govuk-button--secondary lbh-button--secondary"
      style="width: 222px;"
      type="button"
    >
      Close Case
    </button>
  </div>
</div>
`;

exports[`review-documents-view it renders ReviewDocuments correctly on state DocumentsRequestedDes 1`] = `
<div>
  <div
    data-testid="soletojoint-ReviewDocuments"
  >
    <h1
      class="lbh-heading-h1"
    >
      Sole tenant requests a joint tenure
    </h1>
    <h2
      class="lbh-heading-h2 entity-summary__tenure-heading"
    >
      <div>
        Tenure payment ref
         
        mt0xzvjp6g
      </div>
      <div>
        72251 Champlin Trafficway, 99647
      </div>
      <div>
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/person/a302fe28-e5c3-4c84-a3e6-cd575e65093f"
        >
          Global Web Consultant Alverta Denesik
        </a>
      </div>
    </h2>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Eligibility checks passed
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="padding-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735"
        >
          Show all eligibility checks
        </a>
      </div>
    </div>
    <div
      class="mtfh-box mtfh-box--success"
    >
      <div
        class="mtfh-status-heading"
      >
        <svg
          fill="none"
          height="45"
          viewBox="0 0 45 45"
          width="45"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="22.5"
            cy="22.5"
            fill="white"
            r="21.75"
            stroke="#00664F"
            stroke-width="1.5"
          />
          <path
            clip-rule="evenodd"
            d="M35 15.5127L18.775 33L11 24.6201L14.2591 21.1074L18.775 25.9746L31.7409 12L35 15.5127Z"
            fill="#00664F"
            fill-rule="evenodd"
          />
        </svg>
        <h4
          class="lbh-heading-h4 mtfh-status-heading__title"
        >
          Documents requested
        </h4>
      </div>
      <div
        class="govuk-link lbh-link lbh-link--no-visited-state"
        style="margin-left: 60px; margin-top: 17.5px;"
      >
        <a
          class="govuk-link lbh-link lbh-link--no-visited-state"
          href="/processes/soletojoint/e63e68c7-84b0-3a48-b450-896e2c3d7735"
        >
          View request in Document Evidence Store
        </a>
      </div>
    </div>
    <div
      class="govuk-checkboxes__item"
    >
      <input
        class="govuk-checkboxes__input"
        id="condition"
        type="checkbox"
      />
      <label
        class="govuk-label govuk-checkboxes__label"
        for="condition"
      >
        I need to make an appointment to check supporting documents
      </label>
    </div>
    <div
      style="padding-bottom: 35px;"
    />
    <form
      action="#"
      class="review-documents-form"
      id="review-documents-form"
      novalidate=""
    >
      <h4
        class="lbh-heading-h4"
      >
        Use the form below to record the documents you have checked:
      </h4>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          aria-describedby="seen-photographic-id-hint"
          class="govuk-checkboxes__input"
          id="seen-photographic-id"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="seen-photographic-id"
        >
          I confirm that an outcome letter has been sent to the resident
        </label>
        <span
          class="govuk-hint govuk-checkboxes__hint lbh-hint"
          id="seen-photographic-id-hint"
        >
          (for example: valid passport, driving licence)
        </span>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          aria-describedby="seen-second-id-hint"
          class="govuk-checkboxes__input"
          id="seen-second-id"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="seen-second-id"
        >
          I confirm I have seen a second form of ID (does not have to be photographic)
        </label>
        <span
          class="govuk-hint govuk-checkboxes__hint lbh-hint"
          id="seen-second-id-hint"
        >
          (for example: utility bill, bank statement, council letter)
        </span>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          aria-describedby="is-not-immigration-control-hint"
          class="govuk-checkboxes__input"
          id="is-not-immigration-control"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="is-not-immigration-control"
        >
          I confirm that the prospective tenant is not subject to immigration control under the Asylum And Immigration Act 1996
        </label>
        <span
          class="govuk-hint govuk-checkboxes__hint lbh-hint"
          id="is-not-immigration-control-hint"
        >
          (for example: passport, home office letter, embassy letter, immigration status document)
        </span>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          aria-describedby="seen-proof-of-relationship-hint"
          class="govuk-checkboxes__input"
          id="seen-proof-of-relationship"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="seen-proof-of-relationship"
        >
          I confirm I have seen proof of relationship to the existing tenant
        </label>
        <span
          class="govuk-hint govuk-checkboxes__hint lbh-hint"
          id="seen-proof-of-relationship-hint"
        >
          (for example: marriage or civil partner certificate)
        </span>
      </div>
      <div
        class="govuk-checkboxes__item"
      >
        <input
          aria-describedby="incoming-tenant-living-in-property-hint"
          class="govuk-checkboxes__input"
          id="incoming-tenant-living-in-property"
          type="checkbox"
        />
        <label
          class="govuk-label govuk-checkboxes__label"
          for="incoming-tenant-living-in-property"
        >
          I confirm I have seen 3 separate documents proving the proposed tenant has been living at the property for a minimum of 12 months
        </label>
        <span
          class="govuk-hint govuk-checkboxes__hint lbh-hint"
          id="incoming-tenant-living-in-property-hint"
        >
          (for example: letter, utility bill, council tax bill)
        </span>
      </div>
      <button
        class="govuk-button lbh-button"
        disabled=""
        style="width: 222px;"
        type="submit"
      >
        Next
      </button>
    </form>
    <p
      class="lbh-body-m"
    >
      If the documents are not suitable and all avenues to obtain the right documents have been exhausted, then close the case.
    </p>
    <button
      class="govuk-button lbh-button govuk-button--secondary lbh-button--secondary"
      style="width: 222px;"
      type="button"
    >
      Close Case
    </button>
  </div>
</div>
`;
